### PR TITLE
Mja jae issue six

### PR DESCRIPTION
**Pull request by arche4**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

ISSUE SIX

Show an error message if the user tries to submit a new item that has the exact same name as an existing item
Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalization has been normalized and punctuation has been removed

Captura
The user’s original input should be what gets saved in the database

### CODE REVIEW

File: src/components/Product/Product.js [hunk 1]
```diff
@@ -1,43 +1,87 @@
...
+  useEffect(() => {
     firebase
       .firestore()
-      .collection(token())
+      .collection('products')
+      .onSnapshot((snapshot) => {
+        let products = [];
+        snapshot.forEach((doc) => products.push(doc.data().name));
+        setProductData(products);
+      });
```
Comment (alejandronanez) [inline src/components/Product/Product.js]: 👍 I'm glad you figured this out! :D

File: src/components/Product/Product.js [hunk 2]
```diff
@@ -1,43 +1,87 @@
...
+  }, []);
+
+  const validateDuplicate = (product) => {
+    let productDataResult = productData;
+    if (productDataResult.indexOf(product) > -1) {
+      return true;
+    }
+    return false;
```
Comment (alejandronanez) [inline src/components/Product/Product.js]: Nice!

For future occasions, there's no need to do the `let productDataResult`. You could had done it like this

```
if (productData.indexOf(product) > -1) { ... }
```

File: src/components/Product/Product.js [hunk 3]
```diff
@@ -1,43 +1,87 @@
...
       [event.target.name]: event.target.value,
     });
   };
-  var rand = function () {
-    return Math.random().toString(36).substr(2);
-  };
-  var token = function () {
-    return rand() + rand();
+
+  const formate = (name) => {
```
Comment (alejandronanez) [inline src/components/Product/Product.js]: I think you misspelled this work, it should be `format` not `formate` 😬

**Review by alejandronanez — CHANGES_REQUESTED** [review]
The code looks good - nice job 👍.

I'm requesting changes because the AC has not been met completely. We need to store the item name in its original format instead of defaulting to storing it in lowercase.

I explain that in this video: https://www.loom.com/share/ccbc2edb837d4cdcb6803eb322bfe92a

**Comment by Lizhtml** [comment]
Very good work, the code looks excellent. I had fun testing the functionality. ;)

File: src/components/Product/Product.js (hunk 2 above)
Comment (arche4) [reply src/components/Product/Product.js]: Hi Alejo, very thanks, for your comments, is corrected :) <3

File: src/components/Product/Product.js (hunk 3 above)
Comment (arche4) [reply src/components/Product/Product.js]: jejejejej is corrected !!

**Review by alejandronanez — APPROVED** [review]
Nice work @arche4 @AndrewEspejo 

Let's get this merged! 🚜

**Pull request merged**